reark/src/main/java/io/reark/reark/data/stores/cores/ContentProviderStoreCoreBase.java

```diff
@@ -111,33 +117,43 @@ private void initialize() {
         contentResolver.registerContentObserver(getContentUri(), true, getContentObserver());
 
         // Observable transforming updates and inserts to ContentProviderOperations
-        Observable<ContentProviderOperation> operationObservable = updateSubject
+        Observable<CoreOperation> operationObservable = operationSubject
                 .onBackpressureBuffer()
                 .observeOn(Schedulers.io())
-                .flatMap(pair -> createOperation(pair.first, pair.second));
+                .flatMap(this::createCoreOperation);
 
         // Group the operations to a list that should be executed in one batch. The default
         // grouping logic is suitable for pojo stores, but some stores may need to provide
         // their own grouping logic, if for example buffering delays are undesirable.
         updateSubscription = groupOperations(operationObservable)
                 .observeOn(Schedulers.computation())
-                .map(ArrayList::new)
-                .doOnNext(list -> Log.v(TAG, "Grouped list of " + list.size()))
-                .doOnNext(operations -> {
-                    try {
-                        ContentProviderResult[] result = contentResolver.applyBatch(getAuthority(), operations);
-                        Log.v(TAG, String.format("Applied %s operations", result.length));
-                    } catch (RemoteException | OperationApplicationException e) {
-                        Log.e(TAG, "Error applying operations", e);
-                    }
-                })
+                .doOnNext(operations -> Log.v(TAG, "Grouped list of " + operations.size()))
+                .doOnNext(this::applyOperations)
                 .flatMap(Observable::from)
-                .map(ContentProviderOperation::getUri)
-                .subscribe(locker::release,
+                .subscribe(this::release,
                         // On error we can't release the processing lock, as the Uri reference
                         // is lost. It's perhaps better to error out of the subscription than
                         // to leave some of the Uris locked and continue.
-                        error -> Log.e(TAG, "Error while handling data update!", error));
+                        error -> Log.e(TAG, "Error while handling data operations!", error));
+    }
+
+    private void applyOperations(@NonNull List<CoreOperation> operations) {
```

Comment:
Later on in this file you use final for arguments, could you make it consistent in this and the next method?